### PR TITLE
Add commitment scoring, display, and gating

### PR DESCRIPTION
**Pull request by johnpooch**
## What this PR does

Implements the commitment scoring, display, and gating design (D1–D7): a per-turn **commitment** score stored on the player, displayed everywhere their name appears, and used to gate which games they can join and create. This replaces the completed-game reliability model's blindness to first-game abandoners — a player who abandons their very first game is caught during that game.

**Backend**
- **Stored `commitment` field** on `UserProfile` (`high` / `medium` / `low` / `undefined`), recomputed as a side-effect of phase resolution in `PhaseManager.resolve` (wrapped in a savepoint + try/except so a scoring error can never block resolution).
- **Scoring engine** (`user_profile/commitment.py`): NMR rate over the last N = 10 rated order-phases. Rated set excludes private-game, sandbox, and post-civil-disorder phases (only the 2 triggering movement NMRs count) and kicked/deleted-account members. `low` requires rate ≥ 50% **and** ≥ 2 NMRs and short-circuits the "New" floor; `high` ≤ 10%; `undefined` until 10 rated phases exist. Cut-points are the design doc's provisional numbers, pending the D0 calibration deliverable.
- **Backfill**: `python manage.py backfill_commitment` runs the same engine over history for all players (a management command rather than a data migration so it can be re-run if calibration changes the cut-points).
- **Gating**: new binary `Game.commitment_requirement` (`open` / `committed`) alongside `min_reliability`. Open admits everyone except `low`; Committed admits `high` only; no game admits `low`. Enforced at join (`MeetsCommitmentRequirement`), create (only `high` can require Committed; `low` cannot create human games; game-master creators exempt, mirroring the old rule), and `eligible_only` list filtering. Data migration maps existing games: `reliable_only` → Committed, everything else → Open.
- **API**: `commitment` serialized on user profiles and members (masked in anonymous games), `commitment_requirement` on game list/retrieve/create, and a per-viewer `commitment_eligibility` flag (`eligible` / `committed_locked` / `low_locked`) on game payloads so the client doesn't re-derive gating rules. Schema + Orval client regenerated.
- Coexistence: `reliability_tier` / `min_reliability` keep shipping; retiring them is the design's last step (D8), not this PR.

**Frontend**
- Reusable 4-tier `CommitmentBadge` with tap-to-explain popover, shown on the player profile (plus a renamed "Commitment" card with tier row and info) and in-game player info — identically during active play.
- Create game: the stepped "Player Reliability" segmented control is replaced by a **"Committed players only" switch**, disabled with helper text for non-`high` creators and removed on private games.
- Find Games no longer hides ineligible games: locked cards render with a lock button, an inline reason, and an (i) tap-to-reveal explanation; eligible viewers see a "Committed" badge on committed games. Game info's "Player reliability" row becomes a "Commitment" row (Open / Committed players only).
- Two new mock fixtures (`pending-committed`, `pending-committed-locked`) cover the new card states.

**Notes**
- The inline `import` in `PhaseManager._recompute_commitment` breaks a genuine circular import (`phase.models` ↔ `user_profile.commitment`), per the documented exception — the same pattern as the existing inline `game.models` import in that file.
- New shared `set_commitment` test fixture restores the in-memory profile on teardown, because session-scoped test users leak in-memory state across tests after DB rollback.

## Checklist

- [x] This PR does one thing — no unrelated fixes, refactors, or drive-by cleanups bundled in
- [x] For PRs of any significant complexity: I ran `/review-pr` against this PR in Claude Code and addressed (or responded to) its findings
- [x] Tests cover the change
- [ ] Screenshots embedded in the PR description for any visual changes (see CLAUDE.md)

Notes on the checklist:
- Self-review: `/review-pr` couldn't run in this environment (its `gh` CLI isn't wired to the proxied git remote — same limitation noted on #1095), so I did a manual self-review of the full diff: recompute ordering relative to phase completion and CD flagging, savepoint isolation of the recompute hook, the post-CD clamp's consecutive-movement-NMR derivation, gate parity across permission/create/filter paths, and the per-viewer eligibility field on both list and retrieve serializers.
- Tests: full backend suite 1986 passed (scoring-engine edge cases, join/create/filter gating matrix, resolution-hook and exclusion tests); full frontend suite 528 passed (badge tiers + popover, committed switch payload/disabled states, Find Games query change). `npm run lint` and `npx tsc -b --noEmit` clean.
- Screenshots: captured for all five changed surfaces (Find Games locked + Committed cards, player profile badge + Commitment card, in-game player info badges, create-game switch, game-info Commitment row) but this environment has no way to upload GitHub image attachments (`gh api` is restricted to a pinned operation set). They have been delivered to the session owner, who can drag them into this description.

### CODE REVIEW

File: packages/web/src/components/GameCard.tsx [hunk 1]
```diff
@@ -342,6 +365,43 @@ const GameCard: React.FC<GameCardProps> = ({ game, variant, map }) => {
...
+                </TooltipContent>
+              </Tooltip>
+            </div>
+          )}
+
+          {isPending && isCommitmentLocked && (
+            <p className="flex items-center gap-1 text-xs text-muted-foreground">
+              <Lock className="size-3" />
+              {game.commitmentEligibility === "low_locked"
+                ? "Locked — your commitment is Low"
```
Comment (johnpooch) [inline packages/web/src/components/GameCard.tsx]: Use colon instead of emdashes for both of these strings.

File: packages/web/src/components/GameCard.tsx [hunk 2]
```diff
@@ -123,18 +126,38 @@ const GameCard: React.FC<GameCardProps> = ({ game, variant, map }) => {
...
+    game.commitmentEligibility === "low_locked"
+      ? "Your commitment rating is Low, so you can't join human games right now. Your rating is based on your last 10 rated phases."
+      : "This game admits players with High commitment only. Submit orders on time in your games to raise your rating.";
+
+  const joinGameButton =
+    game.canJoin &&
+    (isCommitmentLocked ? (
+      <Popover>
+        <PopoverTrigger asChild>
+          <Button variant="outline" aria-label="Locked">
```
Comment (johnpooch) [inline packages/web/src/components/GameCard.tsx]: Should this button be disabled?

File: packages/web/src/components/GameCard.tsx [hunk 3]
```diff
@@ -123,18 +126,38 @@ const GameCard: React.FC<GameCardProps> = ({ game, variant, map }) => {
...
-      </TooltipContent>
-    </Tooltip>
-  );
+  const isCommitmentLocked =
+    game.commitmentEligibility === "committed_locked" ||
+    game.commitmentEligibility === "low_locked";
+
+  const lockedReason =
+    game.commitmentEligibility === "low_locked"
+      ? "Your commitment rating is Low, so you can't join human games right now. Your rating is based on your last 10 rated phases."
```
Comment (johnpooch) [inline packages/web/src/components/GameCard.tsx]: "join human games right now" -> "join games right now"

File: service/game/serializers.py [hunk 4]
```diff
@@ -254,8 +274,28 @@ class GameRetrieveSerializer(serializers.Serializer):
...
     min_reliability = serializers.CharField(read_only=True)
+    commitment_requirement = serializers.CharField(read_only=True)
+    commitment_eligibility = serializers.SerializerMethodField()
     total_unread_message_count = serializers.SerializerMethodField()
 
+    @extend_schema_field(serializers.ChoiceField(
+        choices=["eligible", "committed_locked", "low_locked"],
+        allow_null=True,
+    ))
+    def get_commitment_eligibility(self, obj):
```
Comment (johnpooch) [inline service/game/serializers.py:282]: Lots of duplication between this and the method above. Maybe add a method to the Game model to encapsulate duplication?
